server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1336,7 +1332,11 @@ public void assignedTablet(KeyExtent extent) {
   }
 
   @Override
-  public Collection<KeyExtent> migrations() {
-    return migrations.keySet();
+  public Set<KeyExtent> migrations() {
+    Set<KeyExtent> migrationsCopy = new HashSet<KeyExtent>();
+    synchronized (migrations) {
+      migrationsCopy.addAll(migrations.keySet());
+    }
+    return migrationsCopy;
```

Comment:
My thinking is that we don't expect the caller to modify it.  Now that a copy is made, its ok if caller modifies.  But if in the future the implementation uses a concurrent map and does not copy, would not want caller of method to modify.  Also the previous implementation didn't expect caller to modify, just read.